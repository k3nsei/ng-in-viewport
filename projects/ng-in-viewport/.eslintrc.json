{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "rules": {
        "@typescript-eslint/no-explicit-any": "off",
        "@typescript-eslint/no-non-null-assertion": "off",
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "inViewport",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "in-viewport",
            "style": "kebab-case"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "rules": {}
    }
  ]
}
