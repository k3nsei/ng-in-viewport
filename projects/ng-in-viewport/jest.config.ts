import type { Config } from 'jest';

const jestConfig: Config = {
  displayName: 'ng-in-viewport',
  preset: 'jest-preset-angular',
  globalSetup: 'jest-preset-angular/global-setup',
  moduleNameMapper: { '^lodash-es$': 'lodash' },
  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
  coverageDirectory: '<rootDir>/../../coverage/ng-in-viewport',
  coverageProvider: 'v8',
  coverageReporters: ['lcovonly', 'text'],
  coverageThreshold: {
    global: {
      branches: 75,
      functions: 75,
      lines: 75,
      statements: 75,
    },
  },
  coveragePathIgnorePatterns: ['/node_modules/', '<rootDir>/src/mocks/'],
};

export default jestConfig;
