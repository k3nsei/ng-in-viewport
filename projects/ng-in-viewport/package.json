{
  "name": "ng-in-viewport",
  "version": "15.0.2",
  "description": "Allows us to check if an element is within the browsers visual viewport",
  "homepage": "https://k3nsei.github.io/ng-in-viewport/",
  "bugs": {
    "url": "https://github.com/k3nsei/ng-in-viewport/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/k3nsei/ng-in-viewport.git"
  },
  "license": "MIT",
  "author": {
    "name": "Piotr StÄ™pniewski",
    "email": "k3nsei.pl@gmail.com",
    "url": "https://github.com/k3nsei"
  },
  "dependencies": {
    "lodash-es": "^4.17.21",
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^14.0.0 || >=15.0.0",
    "@angular/core": "^14.0.0 || >=15.0.0"
  },
  "publishConfig": {
    "provenance": true
  }
}
