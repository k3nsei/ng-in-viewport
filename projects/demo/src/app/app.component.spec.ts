import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';
import { MockProvider } from 'ng-mocks';
import { EMPTY } from 'rxjs';

import { InViewportService } from 'ng-in-viewport';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
  const createComponent = createComponentFactory({
    component: AppComponent,
    providers: [MockProvider(InViewportService, { trigger$: EMPTY })],
  });
  let spectator: Spectator<AppComponent>;
  let component: AppComponent;

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should create component', () => {
    const actual: boolean = component instanceof AppComponent;
    const expected = true;

    expect(actual).toBe(expected);
    expect(spectator.debugElement.nativeElement).toMatchSnapshot();
  });

  it(`should have title`, () => {
    const actual: string = component.title;
    const expected = 'ng-in-viewport demo';

    expect(actual).toBe(expected);
  });

  it('should render title', () => {
    const actual = spectator.query('header h1');
    const expected = 'ng-in-viewport demo';

    expect(actual).toHaveText(expected);
  });
});
