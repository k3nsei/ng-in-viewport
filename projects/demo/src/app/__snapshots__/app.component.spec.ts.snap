// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create component 1`] = `
<div
  id="root0"
>
  <header
    class="app-header"
  >
    <h1
      class="app-header__title"
    >
      ng-in-viewport demo
    </h1>
  </header>
  <main
    class="app-content"
  >
    <invp-example />
  </main>
</div>
`;
